src/slur.cpp

```diff
@@ -432,6 +542,286 @@ void Slur::GetSpannedPointPositions(Doc *doc, const ArrayOfCurveSpannedElements
     */
 }
 
+std::pair<Point, Point> Slur::AdjustCoordinates(
+    Doc *doc, Staff *staff, std::pair<Point, Point> points, int spanningType, curvature_CURVEDIR drawingCurveDir)
+{
+    StemmedDrawingInterface *startStemDrawInterface = dynamic_cast<StemmedDrawingInterface *>(GetStart());
+    StemmedDrawingInterface *endStemDrawInterface = dynamic_cast<StemmedDrawingInterface *>(GetEnd());
+
+    data_STEMDIRECTION startStemDir = STEMDIRECTION_NONE;
+    int startStemLen = 0;
+    if (startStemDrawInterface) {
+        startStemDir = startStemDrawInterface->GetDrawingStemDir();
+        startStemLen = startStemDrawInterface->GetDrawingStemLen();
+    }
+    data_STEMDIRECTION endStemDir = STEMDIRECTION_NONE;
+    int endStemLen = 0;
+    if (endStemDrawInterface) {
+        endStemDir = endStemDrawInterface->GetDrawingStemDir();
+        endStemLen = endStemDrawInterface->GetDrawingStemLen();
+    }
+
+    // Pointers for the start point of the slur
+    Note *startNote = NULL;
+    // Chord *startParentChord = NULL;
+    Chord *startChord = NULL;
+    if (GetStart()->Is(NOTE)) {
```

Comment:
Coding guidelines expect a `this->` when calling `GetStart()`.  However, this we have many calls to `GetStart()` and `GetEnd()` in this method, you might want just to have a `startElement` and a `endElement` local variables.